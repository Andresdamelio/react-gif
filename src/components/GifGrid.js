import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';
const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className='title'>{category}</h3>
      {loading && <p className='paragraph'>cargando...</p>}
      <div className='gifs'>
        {React.Children.toArray(
          images.map((image) => (
            <GifGridItem title={image.title} url={image.url} />
          ))
        )}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
};

export default GifGrid;
