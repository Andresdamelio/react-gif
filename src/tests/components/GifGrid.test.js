import React from 'react';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Probando el comportamiento del componente <GifGrid />', () => {
  const category = 'One Punch';

  test('Probando que el componente <GifGrid /> se renderiza de forma correcta', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de mostrar items cuando se cargan images con useFetchGifs', () => {
    const images = [
      {
        id: 'tJqyalvo9ahykfykAj',
        title: 'rick and morty middle finger GIF',
        url: 'https://media3.giphy.com/media/tJqyalvo9ahykfykAj/giphy.gif?cid=9d23b6ackteuvlgot4bz3fzckqfmb92abkdqiu998sj4im8s&rid=giphy.gif&ct=g'
      },
      {
        id: 'cOKjNdJDbqNCm4n0Jm',
        title: 'Happy Season 1 GIF by Rick and Morty',
        url: 'https://media2.giphy.com/media/cOKjNdJDbqNCm4n0Jm/giphy.gif?cid=9d23b6ackteuvlgot4bz3fzckqfmb92abkdqiu998sj4im8s&rid=giphy.gif&ct=g'
      },
      {
        id: 'gk3R16JhLP8RUka2nD',
        title: 'Count Me In Rick And Morty GIF',
        url: 'https://media3.giphy.com/media/gk3R16JhLP8RUka2nD/giphy.gif?cid=9d23b6ackteuvlgot4bz3fzckqfmb92abkdqiu998sj4im8s&rid=giphy.gif&ct=g'
      },
      {
        id: '3Fdskc7J0timI',
        title: 'rick and morty GIF by Adult Swim',
        url: 'https://media4.giphy.com/media/3Fdskc7J0timI/giphy.gif?cid=9d23b6ackteuvlgot4bz3fzckqfmb92abkdqiu998sj4im8s&rid=giphy.gif&ct=g'
      },
      {
        id: 'l378BzHA5FwWFXVSg',
        title: 'Scared Rick And Morty GIF by Adult Swim',
        url: 'https://media1.giphy.com/media/l378BzHA5FwWFXVSg/giphy.gif?cid=9d23b6ackteuvlgot4bz3fzckqfmb92abkdqiu998sj4im8s&rid=giphy.gif&ct=g'
      }
    ];
    useFetchGifs.mockReturnValue({
      data: images,
      loading: false
    });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.paragraph').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(images.length);
  });
});
