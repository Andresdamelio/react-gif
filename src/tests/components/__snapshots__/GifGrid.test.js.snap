// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el comportamiento del componente <GifGrid /> Debe de mostrar items cuando se cargan images con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="title"
  >
    One Punch
  </h3>
  <div
    className="gifs"
  >
    <GifGridItem
      key=".0"
      title="rick and morty middle finger GIF"
      url="https://media3.giphy.com/media/tJqyalvo9ahykfykAj/giphy.gif?cid=9d23b6ackteuvlgot4bz3fzckqfmb92abkdqiu998sj4im8s&rid=giphy.gif&ct=g"
    />
    <GifGridItem
      key=".1"
      title="Happy Season 1 GIF by Rick and Morty"
      url="https://media2.giphy.com/media/cOKjNdJDbqNCm4n0Jm/giphy.gif?cid=9d23b6ackteuvlgot4bz3fzckqfmb92abkdqiu998sj4im8s&rid=giphy.gif&ct=g"
    />
    <GifGridItem
      key=".2"
      title="Count Me In Rick And Morty GIF"
      url="https://media3.giphy.com/media/gk3R16JhLP8RUka2nD/giphy.gif?cid=9d23b6ackteuvlgot4bz3fzckqfmb92abkdqiu998sj4im8s&rid=giphy.gif&ct=g"
    />
    <GifGridItem
      key=".3"
      title="rick and morty GIF by Adult Swim"
      url="https://media4.giphy.com/media/3Fdskc7J0timI/giphy.gif?cid=9d23b6ackteuvlgot4bz3fzckqfmb92abkdqiu998sj4im8s&rid=giphy.gif&ct=g"
    />
    <GifGridItem
      key=".4"
      title="Scared Rick And Morty GIF by Adult Swim"
      url="https://media1.giphy.com/media/l378BzHA5FwWFXVSg/giphy.gif?cid=9d23b6ackteuvlgot4bz3fzckqfmb92abkdqiu998sj4im8s&rid=giphy.gif&ct=g"
    />
  </div>
</Fragment>
`;

exports[`Probando el comportamiento del componente <GifGrid /> Probando que el componente <GifGrid /> se renderiza de forma correcta 1`] = `
<Fragment>
  <h3
    className="title"
  >
    One Punch
  </h3>
  <p
    className="paragraph"
  >
    cargando...
  </p>
  <div
    className="gifs"
  />
</Fragment>
`;
